sentry.gradle

```diff
@@ -56,80 +56,88 @@ gradle.projectsEvaluated {
+
+              args.addAll(!config.logLevel ? [] : [
+                      "--log-level", config.logLevel      // control verbosity of the output
+              ])
+              args.addAll(!config.flavorAware ? [] : [
+                      "--url", sentryProps.get("defaults.url"),
+                      "--auth-token", sentryProps.get("auth.token")
+              ])
+              args.addAll(["react-native", "gradle",
+                           "--bundle", bundleOutput,           // The path to a bundle that should be uploaded.
+                           "--sourcemap", sourcemapOutput,     // The path to a sourcemap that should be uploaded.
+                           "--release", releaseName,            // The name of the release to publish.
+                           "--dist", versionCode
+              ])
+              args.addAll(!config.flavorAware ? [] : [
+                      "--org", sentryProps.get("defaults.org"),
+                      "--project", sentryProps.get("defaults.project")
+              ])
+
+              project.logger.info("Sentry-CLI arguments: ${args}")
+
+              def osCompatibility = Os.isFamily(Os.FAMILY_WINDOWS) ? ['cmd', '/c', 'node'] : []
+              commandLine(*osCompatibility, *args)
+
+              enabled true
             }
-
-            //
-            // based on:
-            //   https://github.com/getsentry/sentry-cli/blob/master/src/commands/react_native_gradle.rs
-            //
-            def args = [cliExecutable]
-
-            args.addAll(!config.logLevel ? [] : [
-                    "--log-level", config.logLevel      // control verbosity of the output
-            ])
-            args.addAll(!config.flavorAware ? [] : [
-                    "--url", sentryProps.get("defaults.url"),
-                    "--auth-token", sentryProps.get("auth.token")
-            ])
-            args.addAll(["react-native", "gradle",
-                         "--bundle", bundleOutput,           // The path to a bundle that should be uploaded.
-                         "--sourcemap", sourcemapOutput,     // The path to a sourcemap that should be uploaded.
-                         "--release", releaseName            // The name of the release to publish.
-            ])
-            args.addAll(!config.flavorAware ? [] : [
-                    "--org", sentryProps.get("defaults.org"),
-                    "--project", sentryProps.get("defaults.project")
-            ])
-
-            // The names of the distributions to publish. Can be supplied multiple times.
-            versionCodes.each { versionCode -> args.addAll(["--dist", versionCode]) }
-
-            project.logger.info("Sentry-CLI arguments: ${args}")
-
-            def osCompatibility = Os.isFamily(Os.FAMILY_WINDOWS) ? ['cmd', '/c', 'node'] : []
-            commandLine(*osCompatibility, *args)
-
-            enabled true
+          // chain the upload tasks so they run sequentially in order to run
```

Comment:
Chaining the multiple upload tasks to preserve ordering and executing the cleanup task only after all of the upload tasks are complete is the part I'm not too confident in. It works but I imagine there must be a better way to do this.